src/AppInstallerCLICore/Workflows/ManifestComparator.cpp

```diff
@@ -306,15 +314,35 @@ namespace AppInstaller::CLI::Workflow
 
             bool IsFirstBetter(const Manifest::ManifestInstaller& first, const Manifest::ManifestInstaller& second) override
             {
-                // TODO: Current implementation assumes there is only a single installer type requirement. This needs to be updated
-                // once multiple installerType requirements and preferences are accepted.
-                UNREFERENCED_PARAMETER(first);
-                UNREFERENCED_PARAMETER(second);
-                return true;
+                if (m_preference.empty())
+                {
+                    return false;
+                }
+
+                InstallerTypeEnum firstBaseInstallerType = first.BaseInstallerType;
+                InstallerTypeEnum firstEffectiveInstallerType = first.EffectiveInstallerType();
+
+                InstallerTypeEnum secondBaseInstallerType = second.BaseInstallerType;
+                InstallerTypeEnum secondEffectiveInstallerType = second.EffectiveInstallerType();
+
+                for (auto const& preferredInstallerType : m_preference)
+                {
+                    bool isFirstInstallerTypePreferred = (preferredInstallerType == firstBaseInstallerType) || (preferredInstallerType == firstEffectiveInstallerType);
+                    bool isSecondInstallerTypePreferred = (preferredInstallerType == secondBaseInstallerType) || (preferredInstallerType == secondEffectiveInstallerType);
+
+                    if (isFirstInstallerTypePreferred || isSecondInstallerTypePreferred)

```

Comment:
Just thinking through this, it seems to me that it should be an exclusive or here? Or maybe a check for both being preferred types needs to be added?

* If there is no installer type preference - the installers are equally preferred, return false
* If the first installer type is a preferred installer type and the second installer type is not a preferred installer type - first is better, return true
* If the first installer type is not a preferred installer type and the second installer type is a preferred installer type - second is better, return false

Current Implementation - 
* If both installers are of a preferred installer type - the first installer is preferred, return true

Ideal Implementation -
* If both installers are of a preferred installer type - the installers are equally preferred, return false

Reply:
Thanks for catching that mistake. I did not account for when both installers match the preferred installer type. 

I changed the logic such that:
- If both installers match the preferred installer type, the immediately return false since they are equally preferred. 
- If exactly one of these installers has a match, then we return the state of the first installer match.